### PR TITLE
Add completeness (by dataset) chart

### PR DESCRIPTION
**Pull request by samnlindsay**
Motivated by [this PR](https://github.com/moj-analytical-services/data_linking/pull/236) that I repeatedly go back to just for the image that is really useful to understand the available data in the datasets being linked.

## Example with linked data
```
# For linked datasets, and a supplied list of column names
cols = [
    "forename1_std", 
    "surname_std", 
    "dob", 
    "pnc_id", 
    "cro_number", 
    "booking_no_arr", 
    "crn", 
    "asn_arr", 
    "ptiurn_arr", 
    "committal_date_arr", 
    "offence_ho_code_arr", 
    "postcode_unvalidated_arr"
]
linker.completeness_chart(cols=cols)
```
image

## Example with demo dedupe-only
Less usefully, it does work for single datasets, where it is effectively an alternative to `missingness_chart`
```
# Added to 01_Exploratory_analysis.ipynb (splink_demos)
linker.completeness_chart()
```
Screenshot 2022-07-25 at 10 50 25

### CODE REVIEW

File: splink/missingness.py [hunk 1]
```diff
@@ -61,3 +61,42 @@ def missingness_data(linker, input_tablename):
...
+        group by source_dataset
+        order by count(*) desc)
+        """
+        sqls.append(sql)
+
+    sql = " union all ".join(sqls)
+
+    linker._enqueue_sql(sql, "__splink__df_all_column_completeness")
+
+    df = linker._execute_sql_pipeline()
```
Comment (Th368MoJ) [inline splink/missingness.py:100]: We have a wrapper for enqueue + execute if you're not queueing multiple statements.

```suggestion
    df = linker._enqueue_and_execute_sql_pipeline(
        sql, "__splink__df_all_column_completeness"
    )
```

**Review by Th368MoJ — COMMENTED** [review]
LGTM btw. I'll let Robin give the final thumbs up though.

**Comment by RobinL** [comment]
Looks good.

Some suggestions here:
https://github.com/moj-analytical-services/splink/pull/677

And you've forgotten to commit the `completeness.json` chart spec!

**Review by RobinL — CHANGES_REQUESTED** [review]
Looks good.

Some suggestions here:
https://github.com/moj-analytical-services/splink/pull/677

And you've forgotten to commit the `completeness.json` chart spec!

**Comment by samnlindsay** [comment]
> Looks good.
> 
> Some suggestions here: #677
> 
> And you've forgotten to commit the `completeness.json` chart spec!

Thanks, that's better than the (uncommitted) check for source dataset column that I used, and of course you have a method for the right columns :+1:

**Comment by RobinL** [comment]
Looks good now.  Please could you remove the commented out code and then feel free to merge

image

**Review by RobinL — APPROVED** [review]

**Pull request merged**